cpp/include/cuvs/neighbors/brute_force.hpp

```diff
@@ -18,8 +18,11 @@
 
 #include "ann_types.hpp"
 #include <cuvs/neighbors/ann_types.hpp>
+#include <raft/core/copy.hpp>
```

Comment:
good callout - I've moved the implementations for the update_dataset code and the constructors to the src/ directory in the last two commits.

I've left the implementations for the getters in the header file since they are trivial, and can be inlined by the compiler then - but can move to src if you think thats better